assets/js/src/segments/dynamic/dynamic_segments_filters/custom_fields/date.tsx

```diff
@@ -0,0 +1,228 @@
+      <option value="2017-09-01 00:00:00">{MailPoet.I18n.t('september')}</option>
+      <option value="2017-10-01 00:00:00">{MailPoet.I18n.t('october')}</option>
+      <option value="2017-11-01 00:00:00">{MailPoet.I18n.t('november')}</option>
+      <option value="2017-12-01 00:00:00">{MailPoet.I18n.t('december')}</option>
+    </Select>
+  );
+};
+
+const DateYear = ({ onChange, item }: ComponentProps) => {
+  const currentYear = getYear(new Date());
+  useEffect(() => {
+    if ((item.value === undefined) || item.value === '') {
+      onChange(
+        assign(item, {
+          value: `${currentYear}-01-01 00:00:00`,
+          operator: 'equals',
+        })
+      );
+    }
+  }, [currentYear, onChange, item]);
+
+  return (
+    <Grid.CenteredRow>
+      <Select
+        key="select-operator"
+        value={item.operator}
+        onChange={compose([
+          onChange,
+          assign(item),
+          set('operator', __, {}),
+          get('value'),
+          get('target'),
+        ])}
+      >
+        <option value="equals">{MailPoet.I18n.t('equals')}</option>
+        <option value="before">{MailPoet.I18n.t('before')}</option>
+        <option value="after">{MailPoet.I18n.t('after')}</option>
+      </Select>
+      <Select
+        key="select-year"
+        value={item.value}
+        onChange={compose([
+          onChange,
+          assign(item),
+          set('value', __, {}),
+          get('value'),
+          get('target'),
+        ])}
+      >
+        {range(0, 100).map((sub) => (
+          <option
+            value={`${currentYear - sub}-01-01 00:00:00`}
+            key={currentYear - sub}
+          >
+            {currentYear - sub}
+          </option>
+        ))}
+      </Select>
+    </Grid.CenteredRow>
+  );
+};
+
+const convertDateToString = (value: Date): string | undefined => {
+  if (value === null) {
+    return undefined;
+  }
+  return format(value, 'yyyy-MM-dd 00:00:00');
+};
+
+const parseDate = (value: string): Date | undefined => {
+  const date = parseISO(value);
+  if (!isValid(date)) return undefined;
+  return date;
+};
+
+const DateFullDate = ({ onChange, item }: ComponentProps) => {
+  useEffect(() => {
+    if ((item.value === undefined) || item.value === '') {
+      onChange(
+        assign(item, {
+          value: `${format(new Date(), 'yyyy-MM-dd')} 00:00:00`,
+          operator: 'equals',
+        })
+      );
+    }
+  }, [onChange, item]);
+
+  return (
+    <Grid.CenteredRow>
+      <Select
+        key="select-operator"
+        value={item.operator}
+        onChange={compose([
+          onChange,
+          assign(item),
+          set('operator', __, {}),
+          get('value'),
+          get('target'),
+        ])}
+      >
+        <option value="equals">{MailPoet.I18n.t('equals')}</option>
+        <option value="before">{MailPoet.I18n.t('before')}</option>
+        <option value="after">{MailPoet.I18n.t('after')}</option>
+      </Select>
+      <Datepicker
```

Comment:
We render the same Datepicker for both `year_mont_day` and `year_month` date types. The date picker library can be configured as a month picker (see [demo](https://reactdatepicker.com/#example-month-picker)). I guess it would improve UX if we use the month picker for the year_moth date type.

Reply:
@NeosinneR I guess this is a question for you.

Reply:
Good idea, I was not aware of such option. Let's use that instead!